tests/post/steps/test_liveness.py

```diff
@@ -34,58 +35,76 @@ def check_resource_list(host, resource, namespace):
 
 
 @then(parsers.parse(
-    "we can exec '{command}' in the "
-    "'{pod}' pod in the '{namespace}' namespace"))
-def check_exec(host, command, pod, namespace):
+    "we can exec '{command}' in the pod labeled '{label}' "
+    "in the '{namespace}' namespace"))
+def check_exec(host, command, label, namespace):
+    candidates = _get_pods(host, label, namespace)
+    try:
+        pod, = candidates
+    except ValueError:
+        pytest.fail(
+            "Expected one (and only one) pod with label {l}, found {f}".format(
+                l=label, f=len(candidates)
+            )
+        )
+
     cmd = ' '.join([
         'kubectl',
         '--kubeconfig=/etc/kubernetes/admin.conf',
         'exec',
         '--namespace {0}'.format(namespace),
-        pod,
+        pod['metadata']['name'],
         command,
     ])
 
     with host.sudo():
         host.check_output(cmd)
 
 
-@then(
-    parsers.parse(
-        "we have at least {min_pods_count:d} running pod labeled '{label}'"
-    )
-)
+@then(parsers.parse(
+    "we have at least {min_pods_count:d} running pod labeled '{label}'"))
 def count_running_pods(host, min_pods_count, label):
-    cmd = (
-        'kubectl --kubeconfig=/etc/kubernetes/admin.conf '
-        'get pod -l "{label}" --field-selector=status.phase=Running '
-        '--namespace kube-system -o json'
-    ).format(label=label)
-
     # Just after the deployment, the pods may not in running state yet
     # so implement a timeout
     interval = 3
     attempts = 10
 
-    with host.sudo():
-        for _ in range(attempts):
-
-            ret = host.run(cmd)
-            assert ret.rc == 0, ret.stdout
+    for _ in range(attempts):
```

Comment:
Is this kind of pattern (retry an action `N` times, waiting `t` second between each retry until a condition `pred` is true) is repeated?

If yes, it could be worth having a class/function to abstract that (can be done in another PR/debt ticket).

Even without repetition, the abstraction could be worth it, to avoid "polluting" the test function with unrelated logic.

Reply:
Tracked [here](https://github.com/scality/metalk8s/issues/781#issuecomment-476565028).